src/app/loadout-builder/NoSetsFoundExplainer.tsx

```diff
@@ -0,0 +1,388 @@
+            contents: (
+              <>
+                <div key="item" className={styles.modRow}>
+                  <ExoticArmorChoice
+                    lockedExoticHash={lockedExoticHash!}
+                    onClose={() => dispatch({ type: 'removeLockedExotic' })}
+                  />
+                </div>
+                <i key="hint">Consider removing your exotic choice.</i>
+              </>
+            ),
+          });
+        }
+
+        problems.push({
+          id: `badBucket-${bucketHash}`,
+          description: `The ${defs.InventoryBucket[bucketHash].displayProperties.name} slot does not have items that can fit these mods.`,
+          suggestions,
+        });
+      }
+    }
+  }
+
+  const anyStatMinimums = Object.values(statFilters).some((f) => !f.ignored && f.min > 0);
+
+  const elementMayCauseProblems =
+    armorEnergyRules.lockArmorEnergyType !== LockArmorEnergyType.None &&
+    lockedMods.some(
+      (mod) => mod.plug.energyCost && mod.plug.energyCost.energyType !== DestinyEnergyType.Any
+    );
+  const capacityMayCauseProblems =
+    armorEnergyRules.assumeArmorMasterwork !== AssumeArmorMasterwork.All &&
+    (lockedMods.length || anyStatMinimums);
+
+  if (
+    (!alwaysInvalidMods || alwaysInvalidMods.length === 0) &&
+    (elementMayCauseProblems || capacityMayCauseProblems)
+  ) {
+    // If we might have problems assigning bucket specific mods or mods in the
+    // process worker, offer some advice.
+    problems.push({
+      id: 'armorEnergyRestrictions',
+      description:
+        'DIM is restricted in what assumptions it can make about armor energy type and capacity.',
+      suggestions: _.compact([
+        capacityMayCauseProblems && {
+          id: 'assumeMasterworked',
+          contents: (
+            <button
+              key="assumeMasterworked"
+              type="button"
+              className="dim-button"
+              onClick={() =>
+                dispatch({
+                  type: 'assumeArmorMasterworkChanged',
+                  assumeArmorMasterwork: AssumeArmorMasterwork.All,
+                })
+              }
+            >
+              Assume armor is masterworked
+            </button>
+          ),
+        },
+        elementMayCauseProblems && {
+          id: 'allowEnergyChanges',
+          contents: (
+            <button
+              key="allowEnergyChanges"
+              type="button"
+              className="dim-button"
+              onClick={() =>
+                dispatch({
+                  type: 'lockArmorEnergyTypeChanged',
+                  lockArmorEnergyType: LockArmorEnergyType.None,
+                })
+              }
+            >
+              Allow changes to armor elements
+            </button>
+          ),
+        },
+      ]),
+    });
+  }
+
+  // This time we made it to the LO worker process,
+  // so time to investigate why LO could not find a single set.
+  if (processInfo) {
```

Comment:
Something fun I could try is treating these process worker set rejection reasons as a series of steps and computing the ratio of sets that passed each step, and report steps that cause an extraordinarily large percentage of sets to be rejected.